core/templates/components/ck-editor-helpers/ck-editor-4-widgets.initializer.ts

```diff
@@ -205,11 +205,13 @@ export class CkEditorInitializerService {
                   const customEl = that.element.getChild(0).$;
                   customEl[capital.join('') + 'WithValue'] = (
                     htmlEscaperService.objToEscapedJson(
-                      that.data[spec.name] || ''));
+                      that.data[spec.name] !== undefined ?
```

Comment:
Hi @EricZLou -- reading this, the logic seems a bit weird to me. Could you clarify the difference between `that.data[spec.name]` and `that.data.spec` please? What's an example of what this looks like?

Reply:
Oh yes you're right. So `spec` in this case is something like this:
```
{
      "name": "video_id",
      "description": "The Youtube URL or the YouTube id for this video. (The Youtube id is the 11-character string after \"v=\" in the video URL.)",
      "schema": {
        "type": "unicode"
      },
      "default_value": ""
    }
```

And the keys of `that.data` are the same as the `spec.name`-s. So it should definitely be `that.data[spec.name]`.